lib/interface.go

```diff
@@ -21,5 +21,5 @@ type KhanInterface interface {
 	TransferOwnership(context.Context, string, string) (*TransferOwnershipResult, error)
 	UpdateClan(context.Context, *ClanPayload) (*Result, error)
 	UpdatePlayer(context.Context, string, string, interface{}) (*Result, error)
-	SearchClans(context.Context, string) (*SearchClansResult, error)
+	SearchClans(context.Context, string, ...bool) (*SearchClansResult, error)
```

Comment:
I think it is better to avoid using multiple bool arguments, otherwise you may have the following function call:
```
SearchClans(ctx, "", false, true, true, true, false)
```
It is not readable at all.

Reply:
As I mentioned below, it is meant to be a single optional argument, but I agree. I will follow your suggestion to use enums instead.

Reply:
Done.